tree.go

```diff
@@ -1,170 +1,134 @@
 package ipldgit
 
 import (
+	"bufio"
 	"bytes"
-	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
-	"sync"
+	"strconv"
 
-	cid "github.com/ipfs/go-cid"
-	node "github.com/ipfs/go-ipld-format"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
 )
 
-type Tree struct {
-	entries     map[string]*TreeEntry
-	size        int
-	order       []string
-	cid         cid.Cid
-	rawData     []byte
-	rawDataOnce sync.Once
-}
-
-type TreeEntry struct {
-	name string
-	Mode string  `json:"mode"`
-	Hash cid.Cid `json:"hash"`
-}
-
-func (t *Tree) Cid() cid.Cid {
-	return t.cid
-}
-
-func (t *Tree) String() string {
-	return "[git tree object]"
-}
-
-func (t *Tree) GitSha() []byte {
-	return cidToSha(t.cid)
-}
-
-func (t *Tree) Copy() node.Node {
-	out := &Tree{
-		entries: make(map[string]*TreeEntry),
-		cid:     t.cid,
-		size:    t.size,
-		order:   t.order, // TODO: make a deep copy of this
+// DecodeTree fills a NodeAssembler (from `Type.Tree__Repr.NewBuilder()`) from a stream of bytes
+func DecodeTree(na ipld.NodeAssembler, rd *bufio.Reader) error {
+	lstr, err := rd.ReadString(0)
```

Comment:
I've seen this "read number string until null, strip null, parse number" a bunch of times. I think it's enough lines to split it out, like `readNullTerminatedNumber` :)